test/cpp/interop/pre_stop_hook_server.h

```diff
@@ -0,0 +1,56 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#ifndef GRPC_TEST_CPP_INTEROP_PRE_STOP_HOOK_SERVER_H
+#define GRPC_TEST_CPP_INTEROP_PRE_STOP_HOOK_SERVER_H
+
+#include <grpc/support/port_platform.h>
+
+#include <thread>
+
+#include <grpcpp/server.h>
+
+#include "src/core/lib/config/core_configuration.h"
+
+namespace grpc {
+namespace testing {
+
+class PreStopHookServer;
+
+class PreStopHookServerManager {
```

Comment:
Can you add some comments here as to what this does please?

Reply:
Did that.